provisioner/src/lib.rs

```diff
@@ -173,7 +174,13 @@ impl MyProvisioner {
 
             // Make sure database can't see other databases or other users
             // For #557
-            let options = self.pool.connect_options().clone().database(&database_name);
+            let options = self
+                .pool
+                .connect_options()
+                .deref()
+                .clone()
+                .database(&database_name);
```

Comment:
This is how they do it in sqlx's tests: https://github.com/launchbadge/sqlx/blob/main/sqlx-postgres/src/testing/mod.rs#L176-L180